src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java

```diff
@@ -219,11 +219,15 @@ public void keepHeading(double x, double y, boolean fieldRelative) {
    * @param rot Desired angular rate of the robot, [-1,1].
    * @param fieldRelative Whether the provided x and y speeds are relative to the field.
    */
-  public void rotateOrKeepHeading(double x, double y, double rot, boolean fieldRelative) {
+  public void rotateOrKeepHeading(
+      double x, double y, double rot, boolean fieldRelative, double POVangle) {
     // x, y, and rot are all being deadbanded from 0.1 to 0.0, so checking if they're equal to 0
     // does account for controller deadzones.
-    if (x == 0 && y == 0 && rot == 0) {
+    if (x == 0 && y == 0 && rot == 0 && POVangle == -1) {
       setX();
+    } else if (POVangle != -1) {
+      targetHeadingDegrees = POVangle;
```

Comment:
Robot heading is counterclockwise whereas the controller is clockwise, so we need to flip it